internal/alb/tg/targets.go

```diff
@@ -157,3 +164,32 @@ func tdsString(tds []*elbv2.TargetDescription) string {
 	}
 	return strings.Join(s, ", ")
 }
+
+func populateTargetAZ(ctx context.Context, c aws.CloudAPI, a []*elbv2.TargetDescription) error {
+	vpc, err := c.GetVpcWithContext(ctx)
+	if err != nil {
+		return err
+	}
+	cidrBlocks := make([]*net.IPNet, 0)
+	for _, cidrBlockAssociation := range vpc.CidrBlockAssociationSet {
+		_, ipv4Net, err := net.ParseCIDR(*cidrBlockAssociation.CidrBlock)
+		if err != nil {
+			return err
+		}
+		cidrBlocks = append(cidrBlocks, ipv4Net)
+	}
+
+	for i := range a {
+		inVPC := false
+		for _, cidrBlock := range cidrBlocks {
+			if cidrBlock.Contains(net.ParseIP(*a[i].Id)) {
+				inVPC = true
+				break
+			}
+		}
+		if inVPC {
```

Comment:
umm, i suppose it should be `if !inVPC`?

Reply:
Oops, you're right. I'll get that fixed and investigate why the unit tests didn't fail.

Reply:
I figured out why the unit test incorrectly failed and fixed both the test and the conditional in `populateTargetAZ`